src/platform/plugins/shared/data/common/search/aggs/metrics/count.ts

```diff
@@ -48,6 +48,10 @@ export const getCountMetricAgg = () =>
       if (value === 0 && agg.params.emptyAsNull) {
         return null;
       }
+      if (value == null) {
```

Comment:
```suggestion
      if (value == null && !agg.params.emptyAsNull) return 0;
```

Reply:
just a nit, to match how its written in other aggs

Reply:
I've changed to that, but now the function returns `undefined` when there's a time shift instead of `null`, which is inconsistent.
In the original code if the value was either `undefined` or `null` the flag was checked to return consistently either `0` or `null`.